app/algorithms.js

```diff
@@ -36,110 +38,129 @@ module.exports.call = function(websites) {
+          }
+        }, error => {
+          console.error('DateEventExcraction: ' + error);
+        });
+
+      console.log('Call Ollie');
+      ollieQueue.add(() => call(config.algorithms.ollie, replacedCorefs))
+        .then(result => {
+          if (result) {
+            if (typeof(result) === 'string') {
+              console.log('Ollie: Result is a String: ' + result);
             } else {
-              console.log('Ollie: Finished, but result was ' + result);
+              // write to db
+              console.log('Ollie: Write JSON to DB');
+              dbConnection.writeRelationships(result);
             }
-          }, error => {
-            console.error('Ollie: ' + error);
-          });
-        resolve();
-      });
-  });
-}
+          } else {
+            console.log('Ollie: Finished, but result was ' + result);
+          }
+        }, error => {
+          console.error('Ollie: ' + error);
+        });
 
+      console.log('Call OpenIE Stanford');
+      openIeSQueue.add(() => call(config.algorithms.openie_stanford, replacedCorefs))
+        .then(result => {
+          if (result) {
+            if (typeof(result) === 'string') {
+              console.log('OpenIE S: Result is a String: ' + result);
+            } else {
+              // write to db
+              console.log('OpenIE S: Write JSON to DB');
+              dbConnection.writeRelationships(result);
+            }
+          } else {
+            console.log('OpenIE S: Finished, but result was ' + result);
+          }
+        }, error => {
+          console.error('OpenIE S: ' + error);
+        });
 
-function callCoReferenceResolution(data) {
-  return new Promise((resolve, reject) => {
-    const urls = [
-      'http://' + config.algorithms.coreference_resolution.host + ':' + config.algorithms.coreference_resolution.port + '/' + config.algorithms.coreference_resolution.path
-    ];
-    urls.forEach((url) => {
-      request(
-        {
-          url: url,
-          method: 'POST',
-          json: true,
-          headers: {
-            'Content-type': 'application/json',
-          },
-          body: data,
-          timeout: 10000
-        },
-        function callback(error, res) {
-          requestCallback(error, res, resolve, reject);
+      console.log('Call OpenIE Washington');
+      openIeWQueue.add(() => call(config.algorithms.openie_washington, replacedCorefs))
+        .then(result => {
+          if (result) {
+            if (typeof(result) === 'string') {
+              console.log('OpenIE W: Result is a String: ' + result);
+            } else {
+              // write to db
+              console.log('OpenIE W: Write JSON to DB');
+              dbConnection.writeRelationships(result);
+            }
+          } else {
+            console.log('OpenIE W: Finished, but result was ' + result);
+          }
+        }, error => {
+          console.error('OpenIE W: ' + error);
```

Comment:
we have a lot of code duplication here. @Sandr00 and I are working on removing it in a separate PR.